### PR TITLE
Added support to batch jobs.

### PR DESCRIPTION
**Pull request by bwplotka**
Also changing wait Download image check to pre download pull which does the same, but allow us to easily capture output of batch job itself.

Signed-off-by: Bartlomiej Plotka <bwplotka@gmail.com>

### CODE REVIEW

File: env_docker.go [hunk 1]
```diff
@@ -356,6 +348,37 @@ func (d *dockerRunnable) IsRunning() bool {
...
+	}
+
+	d.logger.Log("Starting", d.Name())
+
+	// Make sure the image is available locally; if not wait for it to download.
+	if err = d.prePullImage(ctx); err != nil {
+		return "", err
+	}
+
+	cmd := d.env.execContext(ctx, "docker", append([]string{"run", "-t", "--rm"}, d.env.buildDockerRunArgs(d.name, d.ports, d.opts)...)...)
```
Comment (saswatamcode) [inline env_docker.go:371]: Would this result in double `--rm` flags as we add one in [here](https://github.com/efficientgo/e2e/pull/28/files#diff-ab6d0ab6f93748074333740158654203b796c2ccb2379d33060072557287ff34R234)?

**Review by saswatamcode — APPROVED** [review]
Thanks! 💫

File: env_docker.go (hunk 1 above)
Comment (bwplotka) [reply env_docker.go:371]: uuupd

File: env.go [hunk 2]
```diff
@@ -138,6 +137,9 @@ type runnable interface {
 	// Start tells Runnable to start.
 	Start() error
 
+	// RunOneOff tells Runnable to start as batch job and wait until completion with output capture.
+	RunOneOff(ctx context.Context) (output string, err error)
```
Comment (PhilipGough) [inline env.go:141]: nit: I find the naming confusing here. Suggestions `RunOnce`, `RunOnceOff` or borrowing kube terminology `RunAsJob`

Comment (bwplotka) [reply env.go:141]: I like those, what do you prefer?

**Comment by bwplotka** [comment]
Addressed PTAL (:

**Review by saswatamcode — APPROVED** [review]

File: env.go [hunk 3]
```diff
@@ -138,6 +137,9 @@ type runnable interface {
 	// Start tells Runnable to start.
 	Start() error
 
+	// RunOneOff tells Runnable to start as batch job and wait until completion with output capture.
```
Comment (PhilipGough) [inline env.go:140]:
```suggestion
	// RunOnce tells Runnable to start as batch job and wait until completion with output capture.
```

**Review by PhilipGough — CHANGES_REQUESTED** [review]

**Pull request merged**